ci/jwt/jwks_server.py

```diff
@@ -0,0 +1,130 @@
+        except jwk.InvalidJWKType:
+            key_json = key.export()
+        key_dict = json.loads(key_json, object_pairs_hook=OrderedDict)
+        key_dict['kid'] = key.thumbprint()
+        return key_dict
+
+    def base64_padding(self, value):
+        return value + "=" * (-len(value) % 4)
+
+    def decode_token(self, token):
+        head_b64, payl_b64, sig = token.split(".", 3)
+        head = base64.urlsafe_b64decode(self.base64_padding(head_b64))
+        payl = base64.urlsafe_b64decode(self.base64_padding(payl_b64))
+        head_dict = json.loads(head, object_pairs_hook=OrderedDict)
+        payl_dict = json.loads(payl, object_pairs_hook=OrderedDict)
+        return head_dict, payl_dict
+
+    def do_GET(self): # pylint: disable=C0103
+        logging.info("GET: %s", str(self.path))
+        parts = self.path.strip('/').split('/')
+
+        if len(parts) == 2:
+            key_name, alg = parts[0], parts[1]
+            key = self.get_key(key_name, alg)
+            resp = json.dumps({
+                'keys': [
+                    self.export_key(key)
+                ]
+            })
+        else:
+            key_list = []
+            for key in keys.values():
+                key_list.append(self.export_key(key))
+            resp = json.dumps({
+                'keys': key_list
+            })
+
+        self.reply(resp)
+
+    def do_POST(self): # pylint: disable=C0103
+        logging.info("POST %s", str(self.path))
+
+        content_length = int(self.headers['Content-Length'])
+        post_data = self.rfile.read(content_length)
+        logging.info("BODY %s", post_data.decode('utf-8'))
+
+        key_name, alg = self.path.strip('/').split('/')
+        key = self.get_key(key_name, alg)
+
+        head, payl = self.decode_token(post_data.decode('utf-8'))
+        head['alg'] = alg
+        head['kid'] = key.thumbprint()
+
+        token = jwt.JWT(header=head, claims=payl)
+        token.make_signed_token(key)
+        self.reply(token.serialize())
+
+    def do_DELETE(self): # pylint: disable=C0103
+        logging.info("DELETE %s", str(self.path))
+        parts = self.path.strip('/').split('/')
+        if len(parts) == 2:
+            del keys[parts[0]]
+        else:
+            keys.clear()
+        self.reply("DELETED")
+
+
+def run(port, server_class=HTTPServer, handler_class=JWKSRequestHandler):
+    server_address = ('', port)
+    httpd = server_class(server_address, handler_class)
+    logging.info('Starting JWKS server...')
+    logging.info('Port: %d', port)
+    try:
+        httpd.serve_forever()
+    except KeyboardInterrupt:
+        pass
+    httpd.server_close()
+    logging.info('Stopping server...')
+
+logging.basicConfig(level=logging.INFO)
+parser = argparse.ArgumentParser(description='Lightweight JWKS server')
+parser.add_argument(
+    '-p',
+    '--port',
+    type=int,
+    default=DEFAULT_PORT,
+    help="defines http server port (default: {})".format(DEFAULT_PORT))
```

Comment:
Use fstring in all places you format string

Reply:
Done